pkg/apis/resource/types.go

```diff
@@ -954,6 +990,42 @@ type DeviceRequestAllocationResult struct {
 	// +optional
 	// +featureGate=DRAAdminAccess
 	AdminAccess *bool
+
+	// UsageRestrictedToNode indicates if the usage of an allocation involving this device
+	// has to be limited to exactly the node that was chosen when allocating the claim.
+	//
+	// +optional
+	UsageRestrictedToNode *bool
```

Comment:
I don't remember: do we really need this in the status? Who acts on it there?

If no, remove it?
If yes, add +featureGate.

Reply:
I explain the behavior of the existing DRA.
If a node selector is specified in a ResourceSlice, when devices in this ResourceSlice allocate to a ResourceClaim, the contents of the node selector in this ResourceSlice are copied to ResourceClaim.status.allocation.nodeSelector.

If the label listed in the node selector matches more than one node, the composable DRA controller does not know which node to attach the device to.
That's why we introduced UsageRestrictedToNode. If this parameter is true, the node name is stored in ResourceClaim.status.allocation.nodeSelector. This node name is the node where the scheduler will deploy the pod.

This allows composable DRA to know which node to attach the device to.

Reply:
But does the composable DRA driver need to read the claim.status.allocation.usageRestrictedToNode field for that? I don't think so.

I knows that it needs to do something because of the binding conditions that it has to modified. It knows the node name from the node selector. That's all that it needs.

Reply:
Worded differently: UsageRestrictedToNode is needed in the spec and needs to be handled by the scheduler. It is not needed in the status.

Reply:
I see. You are right. we will delete it from DeviceRequestAllocationResult.